File: app/src/main/java/com/justnik/mirea_navigation_lesson/ui/home/HomeDetailedFragment.java
package com.justnik.mirea_navigation_lesson.ui.home;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import com.justnik.mirea_navigation_lesson.R;
import com.justnik.mirea_navigation_lesson.databinding.FragmentHomeDetailedBinding;


public class HomeDetailedFragment extends Fragment {
    private FragmentHomeDetailedBinding binding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentHomeDetailedBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        setClickListeners();
    }

    private void setClickListeners() {
        binding.bCat.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Navigation.findNavController(view).navigate(R.id.action_homeDetailedFragment_to_catFragment);
            }
        });

        binding.bDog.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Navigation.findNavController(view).navigate(R.id.action_homeDetailedFragment_to_dogFragment);
            }
        });
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        binding = null;
    }
}

File: app/src/main/java/com/justnik/mirea_navigation_lesson/ui/home/HomeDetailedFragment.kt
package com.justnik.mirea_navigation_lesson.ui.home

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import by.kirich1409.viewbindingdelegate.viewBinding
import com.google.android.material.snackbar.Snackbar
import com.justnik.mirea_navigation_lesson.R
import com.justnik.mirea_navigation_lesson.databinding.FragmentHomeDetailedBinding

class HomeDetailedFragment : Fragment(R.layout.fragment_home_detailed) {

    private val binding: FragmentHomeDetailedBinding by viewBinding()

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        setClickListeners()
    }

    private fun setClickListeners(){
        binding.bCat.setOnClickListener {
            //Навигация через action
            findNavController().navigate(R.id.action_homeDetailedFragment_to_catFragment)
        }

        binding.bDog.setOnClickListener {
            //Навигация через destination
            findNavController().navigate(R.id.dogFragment)
        }

    }
}